lib/test_summary_buildkite_plugin/input.rb

```diff
@@ -199,10 +199,34 @@ def details(yaml_lines)
       end
     end
 
+    class CheckStyle < Base
+      def file_contents_to_failures(str)
+        xml = REXML::Document.new(str)
+        xml.elements.enum_for(:each, '//file').flat_map do |file|
+          file_name = file.attribute('name').value
+
+          file.elements.map do |error|
+            severity = error.attribute('severity').value
+            source = error.attribute('source').value
+            line_num = error.attribute('line').value
+            location = "#{file_name}:#{line_num}"
+            message = error.attribute('message').value
+
+            Failure::Structured.new(
+              summary: "[#{severity}] #{source} @ #{location}",
+              message: message,
+              details: location
```

Comment:
`details` is optional and it doesn't really make much sense to include information already in the summary. I'd also reorder the summary to put the location first and maybe push the `source` into the accordion given it's probably the least interesting information. How does this look to you?
```rb
Failure::Structured.new(
  summary: "[#{severity}] #{location}: #{message}"
  details: source
)
```